src/math/p5.Vector.js

```diff
@@ -118,23 +127,63 @@ function vector(p5, fn) {
     // This is how it comes in with createVector()
     // This check if the first argument is a function
     constructor(...args) {
-      let x, y, z;
+      let dimensions = args.length; // TODO: make default 3 if no arguments
+      let values = args.map(arg => arg || 0);
       if (typeof args[0] === 'function') {
         this.isPInst = true;
         this._fromRadians = args[0];
         this._toRadians = args[1];
-        x = args[2] || 0;
-        y = args[3] || 0;
-        z = args[4] || 0;
-        // This is what we'll get with new p5.Vector()
+        values = args.slice(2).map(arg => arg || 0);
+      }
+      if (dimensions === 0) {
+        this.dimensions = 2;
+        this._values = [0, 0, 0];
+
+      } else {
+        this.dimensions = dimensions;
+        this._values = values;
+      }
+    }
+
+    get values() {
+      return this._values
+    }
+
+    set values(newValues) {
+      this._values = newValues.slice()
+    }
+
+    get x() {
+      return this._values[0] || 0
+    }
+
+    setValue(index, value) {
+      if (index < this._values.length) {
+        this._values[index] = value
       } else {
-        x = args[0] || 0;
-        y = args[1] || 0;
-        z = args[2] || 0;
+        p5._friendlyError(
+          'The index parameter is trying to set a value outside the bounds of the vector',
+          'p5.Vector.setValue'
+        );
       }
-      this.x = x;
-      this.y = y;
-      this.z = z;
+    }
+    get y() {
+      return this._values[1] || 0
+    }
+    get z() {
```

Comment:
should we add a `w` value for 4D? or for nD do users will just search values by number?
@limzykenneth

Reply:
I'm ok adding a w personally, since math for 3D stuff uses 4D vectors/matrices quite often. it seems fairly common in other apis too, e.g. native js DOMPoint. And then anything above 4D would use indices

Reply:
Yeah, adding a `w` is fine for me as well. When working on the documentation we need to balance this convention of using `w` and it potentially being unintuitive for those very new to linear algebra, possibly by de-emphaiszing `w` in some way or other solution.